index.js

```diff
@@ -15,6 +15,7 @@ const defaultKeysToValidate = ['params', 'body', 'query', 'user', 'headers', 'tr
 
 const middleware = (joiOptions, options) => (req, res, next) => {
   const validation = req.route.validation
+  const joiOpts = req.route.joi
```

Comment:
How about:
```javascript
const localJoiOptions = req.route.joi || joiOptions
// snip
const result = joi.validate(toValidate, validation, localJoiOptions)
```

Reply:
I think there's a validation missing

```javascript
const localJoiOptions = typeof req.route.joi === 'object' ? req.route.joi : joiOptions
// snip
const result = joi.validate(toValidate, validation, localJoiOptions)
```

Reply:
This is a bit funny :) We could use `joi` to validate `req.route.joi` :)